nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ScriptUtils.java

```diff
@@ -46,23 +44,19 @@
 import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
 
-import org.apache.nifi.annotation.lifecycle.OnStopped;
 import org.apache.nifi.components.AllowableValue;
 import org.apache.nifi.components.PropertyDescriptor;
 import org.apache.nifi.components.ValidationContext;
 import org.apache.nifi.components.ValidationResult;
 import org.apache.nifi.components.Validator;
-import org.apache.nifi.processor.AbstractSessionFactoryProcessor;
 import org.apache.nifi.processor.Relationship;
 import org.apache.nifi.processor.util.StandardValidators;
 import org.apache.nifi.util.StringUtils;
 
 /**
- * This class contains variables and methods common to scripting processors
+ * This class contains variables and methods common to scripting processors, reporting tasks, etc.
  */
-@Stateful(scopes = {Scope.LOCAL, Scope.CLUSTER},
-        description = "Scripts can store and retrieve state using the State Management APIs. Consult the State Manager section of the Developer's Guide for more details.")
-public abstract class AbstractScriptProcessor extends AbstractSessionFactoryProcessor {
+public class ScriptUtils {
```

Comment:
To me it was as "simple" (read: firm logical boundary, not level of work involved) as extracting all state to one class, and all stateless utility methods and constant definitions to another. I am "ok" with this as it is currently structured, in the sense that it works, and I don't think will cause performance/behavior issues. My comments were from the [Martin Fowler](https://stackoverflow.com/a/522907/70465) school of thought (not to throw accusations at @mattyb149). As I find `ExecuteScript` and `InvokeScriptedProcessor` to be incredibly useful and versatile tools in NiFi, I want to ensure that they will have longevity, so someone adding features or tracking something down weeks/months/years in the future will be able to understand the code.